internal/core/worker/universal/worker.go

```diff
@@ -0,0 +1,130 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"time"
+
+	"github.com/freeverseio/laos-universal-node/internal/config"
+	"github.com/freeverseio/laos-universal-node/internal/core/processor/universal"
+	contractDiscoverer "github.com/freeverseio/laos-universal-node/internal/core/processor/universal/discoverer"
+	contractUpdater "github.com/freeverseio/laos-universal-node/internal/core/processor/universal/updater"
+	utils "github.com/freeverseio/laos-universal-node/internal/core/worker"
+	"github.com/freeverseio/laos-universal-node/internal/platform/scan"
+	"github.com/freeverseio/laos-universal-node/internal/platform/state"
+)
+
+type Worker interface {
+	Run(ctx context.Context) error
+}
+
+type worker struct {
+	waitingTime time.Duration
+	processor   universal.Processor
+}
+
+func NewWorker(c *config.Config,
+	client scan.EthClient,
+	scanner scan.Scanner,
+	stateService state.Service,
+	discoverer contractDiscoverer.Discoverer,
+	updater contractUpdater.Updater,
+) Worker {
+	return &worker{
+		waitingTime: c.WaitingTime,
+		processor: universal.NewProcessor(
+			client,
+			stateService,
+			scanner,
+			c,
+			discoverer,
+			updater),
+	}
+}
+
+func (w *worker) Run(ctx context.Context) error {
+	slog.Info("starting universal worker")
+	startingBlock, err := w.processor.GetInitStartingBlock(ctx)
+	if err != nil {
+		return err
+	}
+
+	evoSynced := true
+	lastBlock := startingBlock
+	for {
+		select {
+		case <-ctx.Done():
+			slog.Info("context canceled")
+			return nil
+		default:
+			slog.Debug("executing block range", "startingBlock", startingBlock, "lastBlock", lastBlock, "evoSynced", evoSynced)
+			prevLastBlock, wasEvoSynced, err := w.executeUniversalBlockRange(ctx, evoSynced, startingBlock, lastBlock)
+			if err != nil {
+				var reorgErr universal.ReorgError
+				if errors.As(err, &reorgErr) {
```

Comment:
if the error is not a reorgErr it is not logged

Reply:
@luispdm I will return log of this error. I know that you say we log errors when they appear but it is possible always that we miss some and we just return error. better have the logs at the top also.

Reply:
> if the error is not a reorgErr it is not logged

it is logged everywhere by `executeUniversalBlockRange` and the functions of the `processor`